packages/http/tools/e2e-tests/src/warrior/step-definitions/thenDefinitions.ts

```diff
@@ -17,31 +31,41 @@ async function thenResponseContainsTheCorrectUrlParametersByName(
 
   const warriorWithId: WarriorWithId = (await response.json()) as WarriorWithId;
 
-  assert.ok(warriorWithId.id === '123');
+  assert(warriorWithId.id === '123');
 }
 
-function thenResponseStatusCodeIsSuccessful(
+async function thenResponseContainsTheCorrectBodyData(
   this: InversifyHttpWorld,
   responseAlias?: string,
-): void {
+): Promise<void> {
   const parsedResponseAlias: string = responseAlias ?? 'default';
-
   const response: Response =
     getServerResponseOrFail.bind(this)(parsedResponseAlias);
-  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
-  assert.ok(response.status >= 200 && response.status < 300);
+
+  const warriorCreationResponse: WarriorCreationResponse =
+    (await response.json()) as WarriorCreationResponse;
+
+  assert(warriorCreationResponse.name === 'Samurai');
+  assert(warriorCreationResponse.type === 'Melee');
```

Comment:
Let's use enums instead of magic strings

Reply:
Sure, fixed